Add tests for InplaceTransform

amaarora opened the pull request:
Add tests to explain functionality for InplaceTransform. 

Since, `InplaceTransform` is used for tabular later, one of the tests includes `pd.Series` to fill missing and also an `int` test to show we return original value when we use `InplaceTransform`.

sgugger commented on dev/04_transform.ipynb:
```diff
@@ -864,6 +864,22 @@
     "        return x"
```
<p>Note that you shouldn't return anything: the InplaceTransform does it for you.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fastai/fastai_dev/pull/223/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='60'/>

amaarora commented:
Thanks @sgugger, I've updated PR as per comment and apologies for the delay :) 

Also, I've removed the test for `int`:
```
@A
def encodes(self, x:int): x=x+1
test_eq_type(f(1), 1)
```
As I found above test a little confusing.

jph00 commented:
Thanks!
